Documentation/contributing.rst

```diff
@@ -812,303 +960,370 @@ Have a question about how the tests work or want to chat more about improving th
 
-::
+#. Set up your :ref:`dev_env`
+#. Check the GitHub issues for `good tasks to get started
+   <https://github.com/cilium/cilium/issues?q=is%3Aopen+is%3Aissue+label%3Agood-first-issue>`_.
 
-    $ make -C Documentation html
+Preparing Changes for Submission
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-After this you can browse the updated docs as HTML starting at
-``Documentation\_build\html\index.html``.
+#. Fork the Cilium repository to your own personal GitHub space or request
+   access to a Cilium developer account on Slack
+#. Push your changes to the topic branch in your fork of the repository.
+#. Submit a pull request on https://github.com/cilium/cilium
+  
+   #. Provide a high level description of the PR, feel free to refer to
+      individual commit messages for details. Remember that the commit
+      descriptions will be used for the git revision history, not the PR
+      description text.
+
+#. As you submit the pull request, your PR will automatically be passed through
+   the CI pipeline which will run the following tests:
+  
+   #. Hound: basic ``golang/lint`` static code analyzer. You need to make the
+      puppy happy.
+   #. Jenkins: Will run a series of tests:
+
+      #. Unit tests
+      #. Single node runtime tests
+      #. Multi node Kubernetes tests
+
+#. As part of the submission, GitHub will have requested a review from the
+   respective code owners according to the ``CODEOWNERS`` file in the
+   repository.
+
+   #. Address any feedback received from the reviewers
+   #. You can push individual commits to address feedback and then rebase your
+      branch at the end before merging.
+
+#. Owners of the repository will automatically adjust the labels on the pull
+   request to track its state and progress towards merging.
+#. Once the PR has been reviewed and the CI tests have passed, the PR will be
+   merged by one of the repository owners. In case this does not happen, ping
+   us on Slack.
+
+.. _submit_pr:
 
-Alternatively you can use a Docker container to build the pages:
+Submitting a pull request
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-::
+Contributions must be submitted in the form of pull requests against the github
+repository at: `<https://github.com/cilium/cilium>`_
 
-    $ make render-docs
+Before hitting the submit button, please make sure that the following
+requirements have been met:
 
-This builds the docs in a container and builds and starts a web server with
-your document changes.
+#. Each commit compiles and is functional on its own to allow for bisecting of
+   commits.
+#. All code is covered by unit and/or runtime tests where feasible.
+#. All changes have been tested and checked for regressions by running the
+   existing testsuite against your changes. See the :ref:`testsuite` section
+   for additional details.
+#. All commits contain a well written commit description including a title,
+   description and a ``Fixes: #XXX`` line if the commit addresses a particular
```

Comment:
What about the first 12 chars of commit ID it is fixing + title of the commit that is fixing?

Reply:
added